<a href="https://colab.research.google.com/github/muhamadrefo/ImageClassificationColab/blob/master/NaiveBayesClassifier_3.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Prepare Environment

In [ ]:
from google.colab import drive
drive.mount('/content/drive')

In [4]:
#Importing Packages

import matplotlib.pyplot as plt
import csv
import numpy as np
import pandas as pd
from sklearn import model_selection
from sklearn.model_selection import train_test_split as tt
from sklearn.metrics import classification_report
from sklearn.metrics import confusion_matrix
from sklearn.metrics import accuracy_score

%matplotlib inline

In [101]:
#Filtered Dataset
#%cd /content/drive/My Drive/TestingPlace/PlaceofDatabase/CNN/
#Unfiltered Dataset
%cd /content/drive/My Drive/Batik/

/content/drive/My Drive/Batik


#Reading dataset

In [116]:
#col_names = ["Path", "Dissimilarity", "Contrast", "Homogeneity", "Energy", "Correlation", "ASM" , "Label"]
#datatrain = pd.read_csv('DataTraining.csv', names=col_names)

data_train = pd.read_csv('DataTraining.csv') #DataTraining.csv nor DataValidation.csv
#data_train = pd.read_csv('DataValidation.csv')
pd.set_option('display.max_columns', 100)
data_train.head(15) #display first 15 data

Path  Dissimilarity  Contrast  Homogeneity  \
0    CNN/DataTrain/Cap/T-Cap  (30).jpg       0.657111  0.960802     0.701605   
1   CNN/DataTrain/Cap/T-Cap  (114).jpg       1.193366  2.508668     0.530835   
2    CNN/DataTrain/Cap/T-Cap  (32).jpg       0.550969  0.743234     0.743635   
3    CNN/DataTrain/Cap/T-Cap  (27).jpg       0.614890  0.864450     0.717309   
4     CNN/DataTrain/Cap/T-Cap  (5).jpg       0.523202  0.786015     0.764165   
5    CNN/DataTrain/Cap/T-Cap  (29).jpg       0.598795  0.844671     0.724906   
6    CNN/DataTrain/Cap/T-Cap  (84).jpg       0.629705  1.068646     0.728483   
7    CNN/DataTrain/Cap/T-Cap  (12).jpg       0.449111  0.618634     0.792212   
8    CNN/DataTrain/Cap/T-Cap  (53).jpg       0.999960  2.129484     0.609203   
9    CNN/DataTrain/Cap/T-Cap  (92).jpg       1.203055  2.573451     0.532090   
10   CNN/DataTrain/Cap/T-Cap  (98).jpg       1.145059  2.454997     0.555274   
11   CNN/DataTrain/Cap/T-Cap  (10).jpg       0.595932  0.957880     0.737824   
12   CNN/DataTrain/Cap/T-Cap  (79).jpg       0.577615  0.950893     0.748125   
13    CNN/DataTrain/Cap/T-Cap  (1).jpg       0.394559  0.549067     0.817720   
14   CNN/DataTrain/Cap/T-Cap  (56).jpg       1.182775  2.561699     0.542047   

      Energy  Correlation       ASM Label  
0   0.373753     0.384801  0.139691   Cap  
1   0.186223     0.614433  0.034679   Cap  
2   0.380415     0.534508  0.144715   Cap  
3   0.375527     0.434121  0.141021   Cap  
4   0.472486     0.541780  0.223243   Cap  
5   0.374617     0.446828  0.140338   Cap  
6   0.258461     0.790074  0.066802   Cap  
7   0.547039     0.466384  0.299252   Cap  
8   0.222582     0.644569  0.049543   Cap  
9   0.190674     0.600987  0.036356   Cap  
10  0.181415     0.629564  0.032911   Cap  
11  0.472098     0.504020  0.222876   Cap  
12  0.279615     0.803592  0.078184   Cap  
13  0.545794     0.608097  0.297891   Cap  
14  0.194724     0.571899  0.037918   Cap

In [117]:
# show columns
data_train.columns 

Index(['Path', 'Dissimilarity', 'Contrast', 'Homogeneity', 'Energy',
       'Correlation', 'ASM', 'Label'],
      dtype='object')

In [118]:
data_train

Path  Dissimilarity  Contrast  \
0         CNN/DataTrain/Cap/T-Cap  (30).jpg       0.657111  0.960802   
1        CNN/DataTrain/Cap/T-Cap  (114).jpg       1.193366  2.508668   
2         CNN/DataTrain/Cap/T-Cap  (32).jpg       0.550969  0.743234   
3         CNN/DataTrain/Cap/T-Cap  (27).jpg       0.614890  0.864450   
4          CNN/DataTrain/Cap/T-Cap  (5).jpg       0.523202  0.786015   
..                                      ...            ...       ...   
531  CNN/DataTrain/Tulis/T-Tulis  (154).jpg       0.267497  0.288637   
532  CNN/DataTrain/Tulis/T-Tulis  (124).jpg       0.433196  0.696809   
533  CNN/DataTrain/Tulis/T-Tulis  (181).jpg       0.512212  0.740951   
534  CNN/DataTrain/Tulis/T-Tulis  (148).jpg       0.410094  0.507587   
535  CNN/DataTrain/Tulis/T-Tulis  (167).jpg       0.237348  0.249039   

     Homogeneity    Energy  Correlation       ASM  Label  
0       0.701605  0.373753     0.384801  0.139691    Cap  
1       0.530835  0.186223     0.614433  0.034679    Cap  
2       0.743635  0.380415     0.534508  0.144715    Cap  
3       0.717309  0.375527     0.434121  0.141021    Cap  
4       0.764165  0.472486     0.541780  0.223243    Cap  
..           ...       ...          ...       ...    ...  
531     0.868366  0.470934     0.788305  0.221779  Tulis  
532     0.809534  0.495744     0.811449  0.245762  Tulis  
533     0.766615  0.371361     0.646387  0.137909  Tulis  
534     0.804691  0.438444     0.641707  0.192233  Tulis  
535     0.882495  0.503582     0.796417  0.253595  Tulis  

[536 rows x 8 columns]

# Encoding the Categorial Data

In [119]:
from sklearn.preprocessing import LabelEncoder
le = LabelEncoder()
df = data_train.apply(le.fit_transform)
df.head(15)

Path  Dissimilarity  Contrast  Homogeneity  Energy  Correlation  ASM  \
0    202            402       383          117     172           40  172   
1     17            529       519            2      11          213   11   
2    204            322       273          188     198          118  198   
3    189            380       339          136     180           75  180   
4    223            296       295          239     337          121  337   
5    200            362       328          154     177           82  177   
6    261            390       399          159      60          394   60   
7     23            215       206          315     449           92  449   
8    227            488       489           44      44          241   44   
9    270            530       528            4      17          183   17   
10   276            511       513           22       3          227    3   
11     1            360       381          176     336          102  336   
12   255            344       377          199      87          417   87   
13     0            157       167          379     446          197  446   
14   230            525       526            8      25          140   25   

    Label  
0       0  
1       0  
2       0  
3       0  
4       0  
5       0  
6       0  
7       0  
8       0  
9       0  
10      0  
11      0  
12      0  
13      0  
14      0

# Splitting the Data into Training and Testing Data

In [120]:
X = df.iloc[:, 0:7].values # per array (Path, Diss, Contr, Homog, Energ, Correl, ASM), when start with 0, 'path' included, if start from 1, mean 'path' not included 
Y = df.iloc[:, 7].values # array (Label 'cap' or 'tulis')
X_train, X_test, Y_train, Y_test = tt(X,Y, test_size=0.2, random_state=4)

In [121]:
X_train.shape

(428, 7)

In [122]:
Y.shape

(536,)

In [123]:
X_test.shape

(108, 7)

In [124]:
X.shape

(536, 7)

#Naive Bayes Model Start Here

In [125]:
from sklearn.naive_bayes import GaussianNB
nb = GaussianNB()
nb.fit(X_train,Y_train)
y_pred = nb.predict(X_test)

In [126]:
from sklearn.metrics import classification_report , confusion_matrix , accuracy_score
cm = confusion_matrix(Y_test,y_pred)

In [127]:
print("Confusion Matrix: \n",cm)

Confusion Matrix: 
 [[55  0]
 [ 3 50]]


In [128]:
print("Classification Report: \n", classification_report(Y_test, y_pred))

Classification Report: 
               precision    recall  f1-score   support

           0       0.95      1.00      0.97        55
           1       1.00      0.94      0.97        53

    accuracy                           0.97       108
   macro avg       0.97      0.97      0.97       108
weighted avg       0.97      0.97      0.97       108



In [129]:
print("Accuracy Score: ", accuracy_score(Y_test, y_pred))

Accuracy Score:  0.9722222222222222


# Analytic

using dataset unfiltered
Accuracy Score Training = 0.74
Accuracy Score Validation = 1.0
Updating Index
Accuracy Score Training = 0.97
Accuracy Score Validation = 1.0
Something gone wrong with confussion matrix on Validation, it's doesn't show matrix


using dataset filtered
Accuracy Score Training = 0.77
Accuracy Score Validation = 0.55
Updating Index
Accuracy Score Training = 0.98
Accuracy Score Validation = 1.0